.github/workflows/mypy.yml

```
@@ -0,0 +1,39 @@
+# Workflow to run mypy on select parts of the CPython repo
+name: mypy
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    paths:
+      - "Tools/clinic/**"
+      - ".github/workflows/mypy.yml"
+  workflow_dispatch:
+
+permissions:
+  contents: read
+
+env:
+  PIP_DISABLE_PIP_VERSION_CHECK: 1
+  FORCE_COLOR: 1
+  TERM: xterm-256color  # needed for FORCE_COLOR to work on mypy on Ubuntu, see https://github.com/python/mypy/issues/13817
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
+  cancel-in-progress: true
+
+jobs:
+  mypy:
+    name: Run mypy on Tools/clinic/
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v4
+        with:
+          python-version: "3.11"
```

Do we definitely need 3.11, or can we use a "recent stable" Python version to avoid needing to update later? Currently 3.11. GitHub don't usually update to the next version _immediately_, helps avoid sudden failures of dependencies not being ready.

```suggestion
          python-version: "3.x"
```

---

3.8-3.10 would _probably_ be fine. But mypy uses Python's `ast` module to parse Python source code. I suppose there's a risk that GitHub Actions might be behaving weirdly one day, and give us Python 3.9, which causes mypy to spuriously fail because it can't parse `clinic.py` because we're using some piece of syntax that's new in Python 3.10.

I guess I'd feel more comfortable pinning the minor version here :)

---

`3.x` will never point to anything older. It only points to 3.11 now, will point to 3.12 (and only 3.12) some time after the October release, only 3.13 a year later... Anyway, pinning is fine.

---

(If we had wanted to test multiple Python versions, we'd need to do a matrix like [Bedevere has](https://github.com/python/bedevere/blob/main/.github/workflows/ci.yml#L16-L29).)

---

Okay, if it's guaranteed to never give us <3.11, I'm happy to go with 3.x!

---

> (If we had wanted to test multiple Python versions, we'd need to do a matrix like [Bedevere has](https://github.com/python/bedevere/blob/main/.github/workflows/ci.yml#L16-L29).)

Not necessary right now imo (but see the huge matrix we have for running mypy over at typeshed! https://github.com/python/typeshed/blob/6c54c74a84642faf8ed9834bd1adbc10c1877326/.github/workflows/tests.yml#L82-L118)

---

Agreed one is enough here! 

That's big but not that unusual: a 2D matrix of all supported Python version on three main operating systems = 18 jobs.

Here's a monster 3D matrix = 5xPython x 3xDjango x 5xDRF = 75 jobs!

https://github.com/jazzband/djangorestframework-simplejwt/blob/f298efa860b4ef097d8448e1515c98d40e54a3e4/.github/workflows/test.yml#L17-L20
https://github.com/jazzband/djangorestframework-simplejwt/actions/runs/4854348958